grails-app/services/com/unifina/service/ProductService.groovy

```diff
@@ -12,6 +12,30 @@ class ProductService {
 	ApiService apiService
 	PermissionService permissionService
 	SubscriptionService subscriptionService
+	final Random random = new Random()
+
+	List<Product> relatedProducts(String id) {
+		Product product = Product.get(id)
+		if (product == null) {
+			return new ArrayList<Product>()
+		}
+		// find Product.owner's other products
+		Set<Product> all = new HashSet<Product>(Product.findAllByOwner(product.owner))
+		// find other products from the same category
+		Set<Product> cat = new HashSet<Product>(Product.findAllByCategory(product.category))
```

Comment:
- Result set size could be hard-limited to some sensible max (e.g. 20)
- Unauthenticated fetch - may return products which the user does not have access to! Easiest solution is probably to use `ProductService#list` and pass the category in `ProductListParams`